### ISSUE TITLE
In the action level selector of the uac the nvda does not read the values when they gain focus with the arrows

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#9356]
### Steps to reproduce:
In the start menu look for:
Change User Account Control settings
In the open window walk with tabe until:
levels of notification.
now toggle the slider control with the up and down arrows.
### Actual behavior:
When walking with the arrows the nvda is mute.
### Expected behavior:
When changing the values with the arrows the nvda should read the value currently focused.
#### NVDA installed/portable/running from source:
instaled
#### NVDA version:
alfa 16885
#### Windows version:
17763.348
#### Does the issue still occur after restarting your PC?
yes
#### Have you tried any other versions of NVDA?
no
log:
[uac.txt](https://github.com/nvaccess/nvda/files/2948016/uac.txt)

**Comment by btman16** [comment]
Hello,

I can confirm that this is still present in 2019.2 beta 1.

I believe the bug was introduced with NVDA 2018.4 when an issue was fixed so it could track applications using UIA when another app is busy. Technically, the slider control is built using UIA.

Thanks,

Brandon

**Comment by Adriani90** [comment]
cc: @codeofdusk, @leonardder

**Comment by btman16** [comment]
Hello,

This is still not working in NVDA 2019.2beta2.

Thanks,

Brandon

**Comment by Qchristensen** [comment]
Not a fix, but a part workaround - when on the slider, press the next object key NVDA+numpad 6 / NVDA+shift+right arrow to read the next object which tells you what level the slider is set to.  Use NVDA+numpad 5 / NVDA+shift+o to re-read when moving the slider (Note that using this method re-reading the current object is eeded even when "Focus moves navigator object" is on as moving the slider does not move the navigator object - probably also related to why it doesn't read anything when the slider is changed).

**Comment by ruifontes** [comment]
Still present on NVDA 2019.3 Beta 3 and last Alpha on 30/01/2020

**Comment by btman16** [comment]
Hi,

I was just going to ask about this.
Are there any updates on it?

Thanks,

Brandon

On 1/30/20, ruifontes <notifications@github.com> wrote:
> Still present on NVDA 2019.3 Beta 3 and last Alpha on 30/01/2020
>
> --
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/9356#issuecomment-580455317


-- 
“Be what you are. This is the first step towards becoming better than you are.”
– J. C. Hare & A. W. Hare

**Comment by btman16** [comment]
Hi,

The issue is still present in NVDA 2020.1beta1.
I think that this was broken starting back in 2018.4 by this entry in the changes file:

• NVDA no longer fails to track focus in File Explorer and other applications using UI Automation when another app is busy (such as batch processing audio). (#7345) 

And if we look at the log output for this particular control:
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (20:00:11.472) - MainThread (7896):
Developer info for navigator object:
name: 'Notification level:'
role: ROLE_SLIDER
roleText: None
states: STATE_FOCUSABLE, STATE_FOCUSED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.UIA.UIA object at 0x047F4570>
Python class mro: (<class 'NVDAObjects.UIA.UIA'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'object'>)
description: ''
location: RectLTWH(left=716, top=406, width=26, height=220)
value: '67'
appModule: <'dllhost' (appName 'dllhost', process ID 2448) at address 3bc6d90>
appModule.productName: 'Microsoft® Windows® Operating System'
appModule.productVersion: '10.0.18362.1'
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
windowHandle: 589938
windowClassName: 'msctls_trackbar32'
windowControlID: 0
windowStyle: 1342181914
extendedWindowStyle: 0
windowThreadID: 404
windowText: ''
displayText: ''
UIAElement: <POINTER(IUIAutomationElement) ptr=0x6e07178 at 732350>
UIA automationID: StateSlider
UIA frameworkID: DirectUI
UIA runtimeID: (42, 589938)
UIA providerDescription: [pid:4752,providerId:0x90072 Override(parent link):Nested [pid:2448,providerId:0x90072 Override(parent link):Unidentified Provider (unmanaged:DUI70.dll)]; Main:Nested [pid:2448,providerId:0x90072 Annotation(parent link):Microsoft: Annotation Proxy (unmanaged:UIAutomationCore.dll); Main:Microsoft: MSAA Proxy (unmanaged:UIAutomationCore.dll)]; Hwnd:Microsoft: HWND Proxy (unmanaged:UIAutomationCore.dll)]
UIA className: CCTrackBar
UIA patterns available: RangeValuePattern, LegacyIAccessiblePattern
It is a UIA control.

Thanks,

Brandon

**Comment by Adriani90** [comment]
cc: @josephsl, @feerrenrut, @leonardder

**Comment by Surveyor123** [comment]
it persists for NVDA 2020.3 and it seems it will not be resolved for the next version.

**Comment by btman16** [comment]
Hello,

The issue continues to also persist in NVDA 2020.4beta1.

Thanks,

Brandon

**Comment by TheQuinbox** [comment]
Any updates on this? It still occurs in 2021.1, stable, Windows 10 20h2.
CC @feerrenrut, @josephsl

**Comment by btman16** [comment]
Hi,

This issue can be fixed by putting the following in the Python console:

import UIAHandler
UIAHandler.badUIAWindowClassNames = UIAHandler.badUIAWindowClassNames + ("msctls_trackbar32",)

Thanks,

Brandon

**Comment by seanbudd** [comment]
@btman16 - would you consider opening a PR which adds this to `badUIAWindowClassNames`?

**Comment by btman16** [comment]
Hi,

The only side effect of doing that is that you can no longer object navigate to read the text-based description of the particular level. However, a workaround is that you can use screen review.

Should another solution be looked into or do you still think this is ok?

Thanks,

Brandon

**Comment by tareh7z** [comment]
Any updates on this?  

Still reproducible on [NVDA alpha-53932,ee365d5a](https://download.nvaccess.org/snapshots/alpha/nvda_snapshot_alpha-53909,e9258e8e.exe).

**Comment by seanbudd** [comment]
Has this been reported to Microsoft yet? It seems like we can't fix this without breaking something else accordding to https://github.com/nvaccess/nvda/issues/9356#issuecomment-1735659005

**Comment by btman16** [comment]
Hello,

I believe this is only an issue with NVDA, as Narrator can track this
correctly and report the value as it changes, as well as previous versions
of NVDA before 2018.4. This issue has existed since 2018.4, and I believe
the following change in that release is correlated with this:

NVDA no longer fails to track focus in File Explorer and other applications
using UI Automation when another app is busy (such as batch processing
audio). (#7345)

This issue is still present in the latest versions of NVDA, including
2025.x and alphas.

Thank you very much for your time and I look forward to any input you may
have.

Sincerely,

Brandon Tyson

**Comment by seanbudd** [comment]
@btman16 - this might be only an issue with NVDA, but it appears we are following specs and doing the right thing here. You seemed to suggest that if we change this behaviour we will break something even more important in NVDA. As such, isn't this issue blocked by Microsoft? if so, please report the issue to them
